src/lualib/Error.ts

```diff
@@ -0,0 +1,56 @@
+interface ErrorType<T> extends Function {
+    name: string;
+    new (...args: any[]): T;
+}
+
+function __TS__GetErrorStack(constructor: Function): string {
+    let level = 1;
+    while (true) {
+        const info = debug.getinfo(level, "f");
+        level += 1;
+        if (!info) {
+            // constructor is not in call stack
+            level = 1;
+            break;
+        } else if (info.func === constructor) {
+            break;
+        }
+    }
+    return debug.traceback(undefined, level);
+}
+
+function __TS__GetErrorString(this: void, error: Error): string {
+    return error.message !== "" ? `${error.name}: ${error.message}` : error.name;
+}
+
+function __TS__InitErrorClass<T>(Type: ErrorType<T>, name?: string): any {
+    if (name) {
+        Type.name = name;
+    }
+    return setmetatable(Type, {
+        __call: (_self: any, message: string) => new Type(message),
+    });
+}
+
+Error = __TS__InitErrorClass(
+    class Error {
```

Comment:
Might be a bit cleaner to make `name` in `__TS__InitErrorClass` non-optional and provide `"Error"` there.